kubernetes/resource_kubernetes_persistent_volume_v1_test.go

```diff
@@ -0,0 +1,2000 @@
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesPersistentVolumeV1Exists(resourceName, &conf1),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.claim_ref.#", "1"),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.claim_ref.0.name", name),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.claim_ref.0.namespace", "default"),
+				),
+			},
+			{
+				Config: testAccKubernetesPersistentVolumeV1Config_hostPath_claimRef_withNamespace(name),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesPersistentVolumeV1Exists(resourceName, &conf2),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.claim_ref.0.namespace", name),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.claim_ref.0.name", name),
+					testAccCheckKubernetesPersistentVolumeV1ForceNew(&conf1, &conf2, false),
+				),
+			},
+			{
+				Config: testAccKubernetesPersistentVolumeV1Config_hostPath_claimRef_withPVC(name),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					testAccCheckKubernetesPersistentVolumeV1Exists(resourceName, &conf2),
+					testAccCheckKubernetesPersistentVolumeClaimV1Exists("kubernetes_persistent_volume_claim.test", &conf3),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.claim_ref.0.namespace", name),
+					resource.TestCheckResourceAttr(resourceName, "spec.0.claim_ref.0.name", name),
+					testAccCheckKubernetesPersistentVolumeV1ForceNew(&conf1, &conf2, false),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckKubernetesPersistentVolumeV1ForceNew(old, new *api.PersistentVolume, wantNew bool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		if wantNew {
+			if old.ObjectMeta.UID == new.ObjectMeta.UID {
+				return fmt.Errorf("Expecting new resource for persistent volume %s", old.ObjectMeta.UID)
+			}
+		} else {
+			if old.ObjectMeta.UID != new.ObjectMeta.UID {
+				return fmt.Errorf("Expecting persistent volume UIDs to be the same: expected %s got %s", old.ObjectMeta.UID, new.ObjectMeta.UID)
+			}
+		}
+		return nil
+	}
+}
+
+func waitForPersistentVolumeDeleted(pvName string, poll, timeout time.Duration) error {
+	conn, err := testAccProvider.Meta().(KubeClientsets).MainClientset()
+
+	if err != nil {
+		return err
+	}
+	ctx := context.TODO()
+
+	for start := time.Now(); time.Since(start) < timeout; time.Sleep(poll) {
+		_, err := conn.CoreV1().PersistentVolumes().Get(ctx, pvName, metav1.GetOptions{})
+		if err != nil && apierrs.IsNotFound(err) {
+			return nil
+		}
+	}
+	return fmt.Errorf("Persistent Volume still exists: %s", pvName)
+}
+
+func testAccCheckKubernetesPersistentVolumeV1Destroy(s *terraform.State) error {
+	conn, err := testAccProvider.Meta().(KubeClientsets).MainClientset()
+
+	if err != nil {
+		return err
+	}
+	ctx := context.TODO()
+
+	timeout := 5 * time.Second
+	poll := 1 * time.Second
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "kubernetes_persistent_volume.test" {
```

Comment:
Please, change `"kubernetes_persistent_volume.test"` to `"kubernetes_persistent_volume_v1.test"`.